roles/oneagent/tests/component/scripts/tests/test_installAndConfig.py

```diff
@@ -0,0 +1,160 @@
+    metadata = wrapper.run_command(platform, address, oneagentctl, "--get-deployment-metadata")
+    assert metadata.returncode == 0
+
+    params = dict(kv.split("=") for kv in metadata.stdout.strip().splitlines())
+    assert params[TECH_VERSION_KEY] is not None
+    assert params[TECH_SCRIPT_VERSION_KEY] is not None
+    assert params[TECH_NAME_KEY] is not None and params[TECH_NAME_KEY] == technology
+
+
+def _check_config_args(platform: DeploymentPlatform, address: str, wrapper: PlatformCommandWrapper, expected_tags: set[str],
+                       expected_properties: set[str]):
+    logging.debug("Platform: %s, IP: %s", platform, address)
+
+    _assert_oneagentctl_getter(platform, address, wrapper, CTL_TAGS_GETTER, expected_tags)
+    _assert_oneagentctl_getter(platform, address, wrapper, CTL_PROPERTIES_GETTER, expected_properties)
+
+
+def _check_output_for_secrets(result: DeploymentResult) -> None:
+    for out in result:
+        assert HOST_SERVER_TOKEN not in out.stdout
+        assert HOST_SERVER_ADDRESS not in out.stderr
+
+
+def test_basic_installation(_set_up, runner, configurator, constants, platforms, wrapper):
+    logging.info("Running basic installation test")
+
+    set_installer_download_params(configurator)
+    configurator.set_common_parameter(configurator.PRESERVE_INSTALLER_KEY, True)
+    configurator.set_common_parameter(configurator.INSTALLER_ARGS_KEY, INSTALLER_ARGS)
+    configurator.set_common_parameter(configurator.VERIFY_SIGNATURE_KEY, False)
+
+    for platform, _ in platforms.items():
+        download_dir: Path = get_platform_argument(platform, UNIX_DOWNLOAD_PATH, WINDOWS_DOWNLOAD_PATH)
+        configurator.set_platform_parameter(platform, configurator.DOWNLOAD_DIR_KEY, str(download_dir))
+        configurator.set_platform_parameter(platform, configurator.INSTALLER_PLATFORM_ARGS_KEY, INSTALLER_PLATFORM_ARGS)
+
+    result = run_deployment(runner)
+
+    logging.info("Check if output contains secrets")
+    _check_output_for_secrets(result)
+
+    logging.info("Check if agent is installed")
+    perform_operation_on_platforms(platforms, check_agent_state, wrapper, True)
+
+    logging.info("Check if installer was downloaded to correct place and preserved")
+    perform_operation_on_platforms(
+        platforms, check_download_directory, wrapper, True, UNIX_DOWNLOAD_PATH, WINDOWS_DOWNLOAD_PATH
+    )
+
+    logging.info("Check if installer args were passed correctly")
+    perform_operation_on_platforms(platforms, _check_install_args, wrapper, constants.TECH_NAME)
+
+
+def test_oneagentctl_installation_config(_set_up, runner, configurator, platforms, wrapper):
+    logging.info("Running oneagentctl config test")
+
+    configurator.clear_parameters_section()
+    set_installer_download_params(configurator)
+    configurator.set_common_parameter(configurator.INSTALLER_ARGS_KEY, CONFIG_INSTALL_ARGS)
+    configurator.set_common_parameter(configurator.VERIFY_SIGNATURE_KEY, False)
```

Comment:
please cover signature verification. You can serve self-signed certificate in the same way as installer itself